data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/service/ContentCacheUpdaterService.java

```diff
@@ -0,0 +1,153 @@
+                storeId=contentStoreDb;
+            }
+            if(null != parsedData) {
+                redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), redisConnection);
+                sink.success();
+            }
+        }catch (JedisException ex) {
+            metrics.incCacheErrorCounter();
+            LOGGER.error("", "Exception when adding to dialcode redis cache", ex);
+            redisConnect.resetConnection();
+            try (Jedis redisConn = redisConnect.getConnection(storeId)) {
+                this.redisConnection = redisConn;
+                if (null != parsedData)
+                    redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), redisConnection);
+            }
+        }
+
+    }
+
+    public Map<String, Object> updateDialCodeCache(Map<String, Object> message, ContentCacheUpdaterSink sink) {
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        Map<String, Object> parsedData = null;
+            String contentNode = dialCodeStoreConnection.get(nodeUniqueId);
+            if (contentNode != null) {
+                parsedData = gson.fromJson(contentNode, mapType);
+            } else {
+                parsedData = new HashMap<>();
+            }
+            Map<String, Object> newProperties = extractProperties(message);
+            parsedData.putAll(newProperties);
+            if (parsedData.size() > 0) {
+                parsedData.values().removeAll(Collections.singleton(null));
+                parsedData.values().removeAll(Collections.singleton(""));
+            }
+        return parsedData;
+    }
+
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> updateContentCache(Map<String, Object> message, ContentCacheUpdaterSink sink) {
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        Map<String, Object> parsedData = null;
+        Map<String, List<String>> listTypeFields = new HashMap();
+            String contentNode = contentStoreConnection.get(nodeUniqueId);
+            if (contentNode == null) {
+                parsedData = new HashMap<>();
+            } else {
+                parsedData = gson.fromJson(contentNode, mapType);
+            }
+            Map<String, Object> newProperties = extractProperties(message);
+            for (Map.Entry<String, Object> entry : newProperties.entrySet()) {
+                if (contentModelListTypeFields.contains(entry.getKey())
+                        && entry.getValue() instanceof String
+                        && !((String) entry.getValue()).isEmpty()
+                ) {
+                    String str = (String) entry.getValue();
+                    List<String> value = toList(str);
+                    listTypeFields.put(entry.getKey(), value);
+                    LOGGER.info("", "type cast to List for field: " + entry.getKey() + " ,nodeUniqueId: " + nodeUniqueId);
+                }
+            }
```

Comment:
Move this code into a utility method named convertType(). The utility method needs to handle datetype conversions as well. Also this method is not very efficient. The loop should be on contentModelListTypeFields but not on newProperties